src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkInterface.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Net.NetworkInformation
+{
+    public abstract class NetworkInterface
+    {
+        /// Returns objects that describe the network interfaces on the local computer.
+        public static NetworkInterface[] GetAllNetworkInterfaces()
+        {
+            return SystemNetworkInterface.GetNetworkInterfaces();
+        }
+
+        public static bool GetIsNetworkAvailable()
+        {
+            return SystemNetworkInterface.InternalGetIsNetworkAvailable();
+        }
+
+        public static int LoopbackInterfaceIndex
+        {
+            get
+            {
+                return SystemNetworkInterface.InternalLoopbackInterfaceIndex;
+            }
+        }
+
+        public static int IPv6LoopbackInterfaceIndex
+        {
+            get
+            {
+                return SystemNetworkInterface.InternalIPv6LoopbackInterfaceIndex;
+            }
+        }
+
+        public virtual string Id { get { throw NotImplemented.ByDesignWithMessage(SR.net_PropertyNotImplementedException); } }
```

Comment:
I think this is one of those cases where all of these methods should really be abstract. The class itself IS abstract, though, so maybe there's a good reason for this? Right now, there's no concrete subtypes of this class that don't implement these things. The only implementer is "SystemNetworkInterface", which is the Win32 provider for this info. In the OSX/Linux implementation for this, I'm still planning on overriding them and throwing PlatformNotSupported if they can't be done. So in that case they might as well be abstract, IMO.